pkgs/development/python-modules/steam/default.nix

```
@@ -0,0 +1,59 @@
+{
+  buildPythonPackage,
+  cachetools,
+  fetchFromGitHub,
+  fetchpatch,
+  gevent,
+  gevent-eventemitter,
+  lib,
+  protobuf,
+  pycryptodomex,
+  requests,
+  setuptools,
+  six,
+  vdf,
+}:
+buildPythonPackage rec {
+  pname = "steam";
+  version = "1.4.4";
+
+  src = fetchFromGitHub {
+    owner = "ValvePython";
+    repo = "steam";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-OY04GsX3KMPvpsQl8sUurzFyJu+JKpES8B0iD6Z5uyw=";
+  };
+
+  patches = [
+    # Fixes upstream bug.
+    #
+    # https://github.com/ValvePython/steam/pull/437
+    (fetchpatch {
+      url = "https://github.com/ValvePython/steam/commit/783f023236b2686afbcd2ad124051dc51c20aff0.patch";
+      hash = "sha256-9p52Kjc1TWmFLMr7dMuU6n1MeVC8g0G6hYNgERbMoAM=";
+    })
+  ];
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    cachetools
+    gevent
+    gevent-eventemitter
+    protobuf
+    pycryptodomex
+    requests
+    six
+    vdf
+  ];
```

```suggestion
  dependencies = [
    cachetools
    pycryptodomex
    requests
    six
    vdf
  ];
  
  optional-dependencies = {
    client = [
      gevent
      gevent-eventemitter
      protobuf
    ];
  };
```

https://github.com/ValvePython/steam/blob/dd147db07f6797b787633cb90a8e9222dcb8b2f2/setup.py#L16-L33